arduino-ide-extension/src/browser/auth/authentication-client-service.ts

```diff
@@ -49,7 +49,10 @@ export class AuthenticationClientService
     this.service
       .session()
       .then((session) => this.notifySessionDidChange(session));
+
     this.setOptions();
+    this.service.initAuthSession()
```

Comment:
Note, above this `initAuthSession` call, the `setOptions` is a void, hence notification. It's a "fire and forget" event, so by the time you call `initAutSession` it is not ensured that the options were set or arrived at all.

In the long run, we have to change all similar calls from `foo(arg: any): void` to `foo(arg: any): Promise<void>`, and change the code from this

```ts
this.setOptions();
this.service.initAuthSession()
```

to this:

```ts
```ts
await this.setOptions();
this.service.initAuthSession()
```

//CC @fstasi, @AlbyIanna, @davegarthsimpson, and @francescospissu